include/pybind11/eval.h

```diff
@@ -136,6 +136,14 @@ object eval_file(str fname, object global = globals(), object local = object())
         pybind11_fail("File \"" + fname_str + "\" could not be opened!");
     }
 
+// Python2 API requires this to be encoded to filesystemencoding.
```

Comment:
This looks great.
I see the `env.PY2` is giving us trouble. There is another easy trick, JIC there isn't a better one:
```
if str is not bytes:  # Python 2 is not supported.
```
(We'll find that easily when we clear out Python 2 with the firehose in a few months.)